src/main/java/it/pagopa/pn/deliverypush/action/DigitalWorkFlowRetryHandler.java

```diff
@@ -0,0 +1,190 @@
+        }
+
+        Integer originalRetryNumber = -1;
+        DigitalAddressSourceInt originalAddressSource = null;
+        LegalDigitalAddressInt originalAddressInfo = null;
+
+        if (timelineElement.get().getDetails() instanceof DigitalSendTimelineElementDetails)
+        {
+            // dovrebbe sempre essere instanceof di questo tipo
+            DigitalSendTimelineElementDetails sendDigitalProgressDetailsInt = (DigitalSendTimelineElementDetails) timelineElement.get().getDetails();
+            originalRetryNumber = sendDigitalProgressDetailsInt.getRetryNumber();
+            originalAddressSource = sendDigitalProgressDetailsInt.getDigitalAddressSource();
+            originalAddressInfo = sendDigitalProgressDetailsInt.getDigitalAddress();
+        }
+        else
+        {
+            // caso decisamente strano...loggo ma non tiro errore perchè tanto, se non c'è l'evento non è che ritentando risolve
+            log.error("elapsedExtChannelTimeout Original timelineevent not found, skipping actions iun={} recIdx={}", iun, recIndex);
+        }
+
+        // devo controllare che il timeout scattato sia ancora rilevante.
+        // per capirlo, verifico se l'ultimo evento in ordine cronologico in timeline per recindex, appartiene a retrynumber e addresssource
+        // è il senddigital o senddigitalprogress .
+        // Se lo è, vuol dire che si può equiparare ad una risposta da ext-channel, che va trattata di conseguenza secondo configurazione.
+        if (checkIfEventIsStillValid(iun, recIndex, timelineElement.get()))
+        {
+            // se lo è schedulo gestisco secondo configurazione
+            // EMULO la response proveniente da ext-channel, così poi la logica sarà la stessa
+            digitalWorkFlowExternalChannelResponseHandler.handleExternalChannelResponse(ExtChannelDigitalSentResponseInt.builder()
+                    .eventCode(EventCodeInt.DP10)
+                    .requestId(timelineId)
+                    .iun(iun)
+                    .eventTimestamp(Instant.now())
+                    .status(ExtChannelProgressEventCat.PROGRESS)
+                    .eventDetails("expired timeout")
+                    .build());
+        }
+        else
+        {
+            // salvo cmq in timeline il fatto che ho deciso di non rischedulare i tentativi
+            NotificationInt notification = notificationService.getNotificationByIun(iun);
+            digitalWorkFlowUtils.addDigitalDeliveringProgressTimelineElement(notification,
+                    EventCodeInt.DP10,
+                    recIndex,
+                    originalRetryNumber,
+                    originalAddressInfo,
+                    originalAddressSource,
+                    false,
+                    null,
+                    Instant.now());
+
+            log.info("elapsedExtChannelTimeout Last timelineevent doesn't match original timelineevent source and retrynumber, skipping more actions iun={} recIdx={}", iun, recIndex);
```

Comment:
Unschedulare in caso di notification View, dunque qui non dovrebbe arrivare mai. Mettere log ad error

Reply:
messo log ad error.

Reply:
unschedulare nel caso di notification view è complesso, non avendo la referenza diretta di che evento è eventualmente in corso.
La cosa cmq migliorerebbe solo i casi in cui per qualche motivo, non arriva dal server di pec un OK/KO, che dovrebbero essere rari (e sono cmq già gestiti).